vicodyn/include/cocaine/vicodyn/request_context.hpp

```diff
@@ -15,13 +17,15 @@ class request_context_t: public std::enable_shared_from_this<request_context_t>
         clock_t::time_point when;
     };
 
-    blackhole::logger_t& logger;
-    clock_t::time_point start_time;
-    std::atomic_flag closed;
+    blackhole::logger_t& logger_;
+    clock_t::time_point start_time_;
+    std::atomic_flag closed_;
+
+    synchronized<std::vector<std::shared_ptr<peer_t>>> used_peers_;
+    synchronized<std::vector<checkpoint_t>> checkpoints_;
+    size_t retry_counter_;
 
-    synchronized<std::vector<std::shared_ptr<peer_t>>> used_peers;
-    synchronized<std::vector<checkpoint_t>> checkpoints;
-    size_t retry_counter;
+    synchronized<boost::any> custom_context_;
```

Comment:
It 's better to avoid using of `boost::any` or hide it inside well defined api, not expose it to class users, we should use tool strengths, so strict typing is preferred.

Reply:
And what do you suggest?

Reply:
I've sense that some kind of patterns constructions should help here, but to keep things simple, and not to dive into over-design, `boost::any` would be sufficient.